### PR TITLE
feat: adding pac4j AuthnRequest Embedded signature

### PR DESCRIPTION
**Pull request by jgribonvald**
Provide a way to be able to add `<ds:Signature>` part into `AuthnRequest`.

### CODE REVIEW

**Comment by leleuj** [comment]
+1

The build is broken and it seems related, although the error is strange: 

```java
> Task :api:cas-server-core-api-configuration-model:generateConfigurationMetadata
SLF4J: No SLF4J providers were found.
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#noProviders for further details.
Exception in thread "main" java.lang.IllegalArgumentException: Cant locate class for Collection
	at org.apereo.cas.configuration.metadata.ConfigurationMetadataClassSourceLocator.lambda$locatePropertiesClassForType$3(ConfigurationMetadataClassSourceLocator.java:79)
```

**Comment by jgribonvald** [comment]
After I didn't understood the use of a collection from the interface, a List or a Set could avoid this problem.

File: api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/pac4j/saml/Pac4jSamlClientProperties.java [hunk 1]
```diff
@@ -169,12 +170,41 @@
...
     /**
      * List of attributes requested by the service provider
      * that would be put into the service provider metadata.
      */
     private List<ServiceProviderRequestedAttribute> requestedAttributes = new ArrayList<>();
 
+    /**
+     * Collection of signing signature blacklisted algorithms, if any, to override the global defaults.
+     */
+    private Collection<String> blackListedSignatureSigningAlgorithms = new ArrayList<>();
```
Comment (mmoayyed) [inline api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/pac4j/saml/Pac4jSamlClientProperties.java]: Change to `List` please

File: api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/pac4j/saml/Pac4jSamlClientProperties.java [hunk 2]
```diff
@@ -169,12 +170,41 @@
...
 
+    /**
+     * Whether or not the authnRequest should be signed.
+     */
+    private boolean authnRequestSigned;
+
+    /**
+     * Whether or not the Logout Request sent from the SP should be signed.
+     */
+    private boolean spLogoutRequestSigned;
```
Comment (mmoayyed) [inline api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/pac4j/saml/Pac4jSamlClientProperties.java]: Change to `signServiceProviderLogoutRequest`

File: api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/pac4j/saml/Pac4jSamlClientProperties.java [hunk 3]
```diff
@@ -169,12 +170,41 @@
      */
     private boolean signServiceProviderMetadata;
 
+    /**
+     * Whether or not the authnRequest should be signed.
+     */
+    private boolean authnRequestSigned;
```
Comment (mmoayyed) [inline api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/pac4j/saml/Pac4jSamlClientProperties.java]: Change to `signAuthnRequest`

File: api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/pac4j/saml/Pac4jSamlClientProperties.java (hunk 1 above)
Comment (jgribonvald) [reply api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/pac4j/saml/Pac4jSamlClientProperties.java]: done

File: api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/pac4j/saml/Pac4jSamlClientProperties.java (hunk 2 above)
Comment (jgribonvald) [reply api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/pac4j/saml/Pac4jSamlClientProperties.java]: Done

File: api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/pac4j/saml/Pac4jSamlClientProperties.java (hunk 3 above)
Comment (jgribonvald) [reply api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/pac4j/saml/Pac4jSamlClientProperties.java]: Done

**Comment by jgribonvald** [comment]
@mmoayyed your requested change are applied !

**Pull request merged**